sensor/kubernetes/certrefresh/localscanner_tls_issuer_test.go

```diff
@@ -177,43 +165,32 @@ func TestLocalScannerTLSIssuerProcessMessageKnownMessage(t *testing.T) {
 		},
 	}
 
-	go func() {
-		assert.NoError(t, fixture.tlsIssuer.ProcessMessage(msg))
-		processMessageDoneSignal.Signal()
-	}()
+	done := make(chan struct{})
+	fixture.certRequester.On("DispatchResponse",
+		certificates.NewResponseFromLocalScannerCerts(expectedResponse)).Run(func(args mock.Arguments) {
+		close(done)
+	}).Once().Return()
+
+	assert.NoError(t, fixture.tlsIssuer.ProcessMessage(msg))
 
 	select {
-	case <-ctx.Done():
-		assert.Fail(t, ctx.Err().Error())
-	case response := <-fixture.tlsIssuer.msgFromCentralC:
-		protoassert.Equal(t, expectedResponse, response)
+	case <-done:
+		fixture.certRequester.AssertExpectations(t)
+	case <-time.After(5 * time.Second):
+		t.Fatalf("Test timed out waiting for DispatchResponse to be called")
 	}
-
-	_, ok := processMessageDoneSignal.WaitWithTimeout(100 * time.Millisecond)
-	assert.True(t, ok)
 }
 
 func TestLocalScannerTLSIssuerProcessMessageUnknownMessage(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
-	defer cancel()
-	processMessageDoneSignal := concurrency.NewErrorSignal()
 	fixture := newLocalScannerTLSIssuerFixture(fakeK8sClientConfig{})
 	msg := &central.MsgToSensor{
 		Msg: &central.MsgToSensor_ReprocessDeployments{},
 	}
 
-	go func() {
-		assert.NoError(t, fixture.tlsIssuer.ProcessMessage(msg))
-		processMessageDoneSignal.Signal()
-	}()
+	assert.NoError(t, fixture.tlsIssuer.ProcessMessage(msg))
 
-	select {
-	case <-ctx.Done():
-	case <-fixture.tlsIssuer.msgFromCentralC:
-		assert.Fail(t, "unknown message is not ignored")
-	}
-	_, ok := processMessageDoneSignal.WaitWithTimeout(100 * time.Millisecond)
-	assert.True(t, ok)
+	time.Sleep(100 * time.Millisecond)
+	fixture.certRequester.AssertNotCalled(t, "DispatchResponse", mock.Anything)
```

Comment:
Let's avoid sleeps in test as they are flake magnets. We can use `assert.Eventually` instead.

Reply:
I think that `assert.Never` works better here.